godot-google-play-billing/src/main/java/org/godotengine/godot/plugin/googleplaybilling/utils/GooglePlayBillingUtils.java

```diff
@@ -51,33 +52,59 @@ public static Dictionary convertPurchaseToDictionary(Purchase purchase) {
+		dictionary.put("products", products);
 		dictionary.put("is_acknowledged", purchase.isAcknowledged());
 		dictionary.put("is_auto_renewing", purchase.isAutoRenewing());
 		return dictionary;
 	}
 
-	public static Dictionary convertSkuDetailsToDictionary(SkuDetails details) {
+	public static Dictionary convertProductDetailsToDictionary(ProductDetails details) {
 		Dictionary dictionary = new Dictionary();
-		dictionary.put("sku", details.getSku());
+		dictionary.put("id", details.getProductId());
 		dictionary.put("title", details.getTitle());
 		dictionary.put("description", details.getDescription());
-		dictionary.put("price", details.getPrice());
-		dictionary.put("price_currency_code", details.getPriceCurrencyCode());
-		dictionary.put("price_amount_micros", details.getPriceAmountMicros());
-		dictionary.put("free_trial_period", details.getFreeTrialPeriod());
-		dictionary.put("icon_url", details.getIconUrl());
-		dictionary.put("introductory_price", details.getIntroductoryPrice());
-		dictionary.put("introductory_price_amount_micros", details.getIntroductoryPriceAmountMicros());
-		dictionary.put("introductory_price_cycles", details.getIntroductoryPriceCycles());
-		dictionary.put("introductory_price_period", details.getIntroductoryPricePeriod());
-		dictionary.put("original_price", details.getOriginalPrice());
-		dictionary.put("original_price_amount_micros", details.getOriginalPriceAmountMicros());
-		dictionary.put("subscription_period", details.getSubscriptionPeriod());
-		dictionary.put("type", details.getType());
+		dictionary.put("type", details.getProductType());
+
+		ProductDetails.OneTimePurchaseOfferDetails oneTimePurchaseOfferDetails = details.getOneTimePurchaseOfferDetails();
+		if (oneTimePurchaseOfferDetails != null) {
+			Dictionary oneTimePurchaseDetailsDictionary = new Dictionary();
+			oneTimePurchaseDetailsDictionary.put("formatted_price", oneTimePurchaseOfferDetails.getFormattedPrice());
+			oneTimePurchaseDetailsDictionary.put("price_amount_micros", oneTimePurchaseOfferDetails.getPriceAmountMicros());
+			oneTimePurchaseDetailsDictionary.put("price_currency_code", oneTimePurchaseOfferDetails.getPriceCurrencyCode());
+			dictionary.put("one_time_purchase_details", oneTimePurchaseDetailsDictionary);
+		}
+
+		List<ProductDetails.SubscriptionOfferDetails> subscriptionOfferDetailsList = details.getSubscriptionOfferDetails();
+
+		if (subscriptionOfferDetailsList != null && !subscriptionOfferDetailsList.isEmpty()) {
+			ArrayList<Dictionary> subscriptionOfferDetailsDictionaryList = new ArrayList<>();
+
+			for (ProductDetails.SubscriptionOfferDetails subscriptionOfferDetails : subscriptionOfferDetailsList) {
+				Dictionary subscriptionOfferDetailsDictionary = new Dictionary();
+				subscriptionOfferDetailsDictionary.put("offer_token", subscriptionOfferDetails.getOfferToken());
+				subscriptionOfferDetailsDictionary.put("offer_tags", subscriptionOfferDetails.getOfferTags());
```

Comment:
```suggestion
				Object[] array = subscriptionOfferDetails.getOfferTags().toArray();
				subscriptionOfferDetailsDictionary.put("offer_tags", array);
```
ArrayList won't work in a Dictionary (it will come to Godot as `null`). It needs to be converted to array.